lib/sparrow/h2_worker/pool.ex

```diff
@@ -87,7 +87,8 @@ defmodule Sparrow.H2Worker.Pool do
   @spec start_link(Sparrow.H2Worker.Pool.Config.t(), pool_type, [atom]) ::
           {:error, any} | {:ok, pid}
```

Comment:
Is the spec correct or is it too wide?
When will `{:error, any}` be returned?

Reply:
Now never because we pattern match against {:ok, pid}